sane_backends: 1.2.1 x86 build review

Ref: https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/4339/builds/80667.log

Haiku R1B4 - hrev56578_69 x86 (GCC 11.2.0)
- Migrate recipe to Python 3.9
- Build review:
```
/bin/sh ../libtool --silent  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../include/sane  -D_REENTRANT -DPATH_SANE_CONFIG_DIR=/packages/sane_backends_x86-1.2.1-1/.settings/sane.d           -DPATH_SANE_DATA_DIR=/packages/sane_backends_x86-1.2.1-1/.self/data           -DPATH_SANE_LOCK_DIR=/packages/sane_backends_x86-1.2.1-1/.self/var/lock           -DV_MAJOR=1 -DV_MINOR=2 -I. -I. -I../include -I../include -I/packages/libusb_x86-1.0.26-2/.self/develop/headers/x86/libusb-1.0 -DLIBDIR="\"/packages/sane_backends_x86-1.2.1-1/.self/lib/x86/sane\"" -DBACKEND_NAME=as6e  -Wall -g -O2 -MT libas6e_la-as6e.lo -MD -MP -MF .deps/libas6e_la-as6e.Tpo -c -o libas6e_la-as6e.lo `test -f 'as6e.c' || echo './'`as6e.c
as6e.c: In function 'as6e_open':
as6e.c:587:21: warning: format '%d' expects argument of type 'int', but argument 3 has type 'pid_t' {aka 'long int'} [-Wformat=]
 587 |           DBG (1, "%d - read %d status = %d\n", getpid (), data_processed,
     |                    ~^                           ~~~~~~~~~
     |                     |                           |
     |                     int                         pid_t {aka long int}
     |                    %ld
```

```
/bin/sh ../libtool --silent  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../include/sane  -D_REENTRANT -DPATH_SANE_CONFIG_DIR=/packages/sane_backends_x86-1.2.1-1/.settings/sane.d           -DPATH_SANE_DATA_DIR=/packages/sane_backends_x86-1.2.1-1/.self/data           -DPATH_SANE_LOCK_DIR=/packages/sane_backends_x86-1.2.1-1/.self/var/lock           -DV_MAJOR=1 -DV_MINOR=2 -I. -I. -I../include -I../include -I/packages/libusb_x86-1.0.26-2/.self/develop/headers/x86/libusb-1.0 -DLIBDIR="\"/packages/sane_backends_x86-1.2.1-1/.self/lib/x86/sane\"" -DBACKEND_NAME=dell1600n_net  -Wall -g -O2 -MT libdell1600n_net_la-dell1600n_net.lo -MD -MP -MF .deps/libdell1600n_net_la-dell1600n_net.Tpo -c -o libdell1600n_net_la-dell1600n_net.lo `test -f 'dell1600n_net.c' || echo './'`dell1600n_net.c
In file included from /boot/system/develop/headers/x86/tiffio.h:31,
                from dell1600n_net.c:73:
/boot/system/develop/headers/x86/tiff.h:83:38: error: conflicting types for 'int32'; have 'int32_t' {aka 'int'}
  83 | typedef TIFF_MSC_DEPRECATED int32_t  int32  TIFF_GCC_DEPRECATED;
     |                                      ^~~~~
In file included from /boot/system/develop/headers/os/support/ByteOrder.h:11,
                from /boot/system/develop/headers/bsd/endian.h:16,
                from /boot/system/develop/headers/posix/netinet/in.h:10,
                from dell1600n_net.c:69:
/boot/system/develop/headers/os/support/SupportDefs.h:24:41: note: previous declaration of 'int32' with type 'int32' {aka 'long int'}
  24 | typedef __haiku_int32                   int32;
     |                                         ^~~~~
In file included from /boot/system/develop/headers/x86/tiffio.h:31,
                from dell1600n_net.c:73:
/boot/system/develop/headers/x86/tiff.h:84:38: error: conflicting types for 'uint32'; have 'uint32_t' {aka 'unsigned int'}
  84 | typedef TIFF_MSC_DEPRECATED uint32_t uint32 TIFF_GCC_DEPRECATED;
     |                                      ^~~~~~
In file included from /boot/system/develop/headers/os/support/ByteOrder.h:11,
                from /boot/system/develop/headers/bsd/endian.h:16,
                from /boot/system/develop/headers/posix/netinet/in.h:10,
                from dell1600n_net.c:69:
/boot/system/develop/headers/os/support/SupportDefs.h:25:41: note: previous declaration of 'uint32' with type 'uint32' {aka 'long unsigned int'}
  25 | typedef __haiku_uint32                  uint32;
     |                                         ^~~~~~
Makefile:5541: recipe for target 'libdell1600n_net_la-dell1600n_net.lo' failed
make[3]: *** [libdell1600n_net_la-dell1600n_net.lo] Error 1
make[3]: Leaving directory '/sources/sane-backends-1.2.1/backend'
Makefile:4267: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/sources/sane-backends-1.2.1/backend'
Makefile:608: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/sources/sane-backends-1.2.1'
Makefile:498: recipe for target 'all' failed
make: *** [all] Error 2
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 2.
Error: Build has failed - stopping.
command exit status: 1
```

--- a/media-gfx/sane_backends/sane_backends-1.2.1.recipe
+++ b/media-gfx/sane_backends/sane_backends-1.2.1.recipe
@@ -16,18 +16,20 @@ documentation."
 HOMEPAGE="http://www.sane-project.org"
 COPYRIGHT="David Mosberger-Tang, Andy Beck"
 LICENSE="GNU LGPL v2"
-REVISION="1"
+REVISION="2"
 SOURCE_URI="https://gitlab.com/sane-project/backends/uploads/110fc43336d0fb5e514f1fdc7360dd87/sane-backends-$portVersion.tar.gz"
 CHECKSUM_SHA256="f832395efcb90bb5ea8acd367a820c393dda7e0dd578b16f48928b8f5bdd0524"
 SOURCE_DIR="sane-backends-$portVersion"
 PATCHES="sane_backends-$portVersion.patchset"
 
-ARCHITECTURES="all ?x86_gcc2"
+ARCHITECTURES="all !x86_gcc2"
 SECONDARY_ARCHITECTURES="x86"
 
 GLOBAL_WRITABLE_FILES="
 	settings/sane.d directory keep-old
 	"
+
+pythonVersion=3.9
 
 PROVIDES="
 	sane_backends$secondaryArchSuffix = $portVersion
@@ -44,7 +46,7 @@ REQUIRES="
 	lib:libcurl$secondaryArchSuffix
 	lib:libjpeg$secondaryArchSuffix
 	lib:libpng16$secondaryArchSuffix
-	lib:libpython3.7m$secondaryArchSuffix
+	lib:libpython$pythonVersion$secondaryArchSuffix
 	lib:libtiff$secondaryArchSuffix
 	lib:libusb_1.0$secondaryArchSuffix
 	lib:libxml2$secondaryArchSuffix
@@ -63,7 +65,7 @@ BUILD_REQUIRES="
 	devel:libcurl$secondaryArchSuffix
 	devel:libjpeg$secondaryArchSuffix
 	devel:libpng16$secondaryArchSuffix
-	devel:libpython3.7m$secondaryArchSuffix
+	devel:libpython$pythonVersion$secondaryArchSuffix
 	devel:libtiff$secondaryArchSuffix
 	devel:libusb_1.0$secondaryArchSuffix
 	devel:libxml2$secondaryArchSuffix
@@ -87,6 +89,10 @@ BUILD_PREREQUIRES="
 
 BUILD()
 {
+	if [ $effectiveTargetArchitecture == 'x86' ]; then
+		export CFLAGS="-DTIFF_DISABLE_DEPRECATED"
+	fi
+
 	export LIBS="-lnetwork"
 	runConfigure configure \
 		--prefix=$prefix \
@@ -104,7 +110,7 @@ BUILD()
 		--enable-rpath \
 		--enable-pnm-backend
 
-	make
+	make $jobArgs
 }
 
 INSTALL()

